pkg/generate/app/componentref.go

```diff
@@ -299,6 +302,9 @@ type ComponentInput struct {
 
 	ExpectToBuild bool
 	ScratchImage  bool
+	// For keeping track of if this image was retrieved from a Dockerfile,
+	// which changes our BuildConfig results
```

Comment:
I'm a bit confused, is this about a Dockerfile or about a reference to a Docker image?

In `oc new-app php~https://github.com/org/repo`, `php` refers to a Docker image, not a Dockerfile.